tasks/sync-granule/tests/sync_granule_test.js

```diff
@@ -40,18 +39,23 @@ async function prepareS3DownloadEvent(t) {
   t.context.event.input.granules[0].files[0].path = granuleFilePath;
   t.context.event.config.fileStagingDir = randomString();
 
-  await validateConfig(t, t.context.event.config);
-  await validateInput(t, t.context.event.input);
-
   await s3().createBucket({ Bucket: t.context.event.config.provider.host }).promise();
 
   // Stage the file that's going to be downloaded
-  const key = `${granuleFilePath}/${granuleFileName}`;
-  await s3().putObject({
-    Bucket: t.context.event.config.provider.host,
-    Key: key,
-    Body: streamTestData(`granules/${granuleFileName}`)
-  }).promise();
+  for (let i = 0; i < t.context.event.input.granules.length; i += 1) {
+    for (let j = 0; j < t.context.event.input.granules[i].files.length; j += 1) {
+      t.context.event.input.granules[i].files[j].path = granuleFilePath;
+      const granuleFileName = t.context.event.input.granules[i].files[j].name;
+      const key = `${granuleFilePath}/${granuleFileName}`;
+
+      // eslint-disable-next-line no-await-in-loop
+      await s3().putObject({
```

Comment:
You can use `aws.s3PutObject()`  here for better stacktraces. If so, you'll want to delete `.promise()` at the end as well.

Reply:
This also applies to the rest of this test file. I think there's only one other instance of `s3().putObject()`, but there's also a `common/aws.s3PutObjectTagging` with better stacktraces than `s3().putObjectTagging`, though it takes separate params instead of a params object.